### DISCUSSION TITLE
TableStructureRec纠正角度后如何避免截取部分有用信息

### DISCUSSION THREAD

**Discussion by Dkaaaaa** [RapidAI/TableStructureRec#81] [Q&A]
您好，我在使用TableStructureRec的时候，发现在纠正角度后，会对原来的图片进行截取，因为我的表格图片留白较少，这会导致我后面的识别就不准确了。
我想知道有什么参数可以调整为只旋转不截图吗或者调整截图的力度？
或者有没有可以扩展图片的方法，那这样是不是就避免把有用信息截取掉的问题呢？
或者只对有角度的表格进行纠正，也就是不会进行旋转截图了，是否支持这种用法呀？
期待您的回复，谢谢！

**Comment by Joker1212** [comment]
> 您好，我在使用TableStructureRec的时候，发现在纠正角度后，会对原来的图片进行截取，因为我的表格图片留白较少，这会导致我后面的识别就不准确了。 我想知道有什么参数可以调整为只旋转不截图吗或者调整截图的力度？ 或者有没有可以扩展图片的方法，那这样是不是就避免把有用信息截取掉的问题呢？ 或者只对有角度的表格进行纠正，也就是不会进行旋转截图了，是否支持这种用法呀？ 期待您的回复，谢谢！

只修正角度的话，参考小角度修正，如果文档角度更大的话，也可以走RapidOritation先修正大角度，再利用openCV小角度修正，RapidTableDetection因为是边缘检测，确实会把表格的部分截取出来

**Reply by Dkaaaaa** [reply]
我的使用场景都是垂直表格，小角度修正尝试过不行。

**Reply by Joker1212** [reply]
请问能给一张去敏的图吗？就是把敏感信息打马赛克，如果只有小角度问题，没有透视的话，emmm,建议直接问下gpt，因为opencv对场景一致的情况下进行调整效果还是可以的，还有一个比较麻烦的办法就是RapidTableDetection其实是输出了边缘的四个角点的，可以看看能不能利用这个找到透视变换矩阵，让整张图按这个角度和透视进行变换

**Reply by Dkaaaaa** [reply]
您好，我讲述的不清楚让您误解了，我的图片都是旋转270度或者90度的，这种情况用您说的小角度修正是不行的，尝试用了您推荐的RapidOritation确定旋转角度，再利用openCV修正，成功啦，谢谢
